### PR TITLE
Doc: Tutorial for re-writing a kubernetes replicaset-controller in Java

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

File: docs/java-controller-tutorial-rewrite-rs-controller.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+
+### Setup Environment
+
+Please check that you have the following required development toolings installed in your local
+environment:
+
+- __JDK 8+__: Necessary Java development environment.
+- __Maven__: Greater than 3.0.0+.
+- __KinD__: (Optional) For verifying the controller in a real kubenretes cluster. https://kind.sigs.k8s.io/docs/user/quick-start/
```
Comment (brendandburns) [inline docs/java-controller-tutorial-rewrite-rs-controller.md:18]: kubernetes (typo)

File: docs/java-controller-tutorial-rewrite-rs-controller.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
+# Re-Write a Kubernetes Replicas-Controller in Java
+
+__TL;DR__: By this document, you will experience how to re-write a replicaset-controller
```
Comment (brendandburns) [inline docs/java-controller-tutorial-rewrite-rs-controller.md:3]: nit: "This tutorial will show you how to re-write a ..."

**Comment by brendandburns** [comment]
Couple of small comments, but otherwise looks great!

File: docs/java-controller-tutorial-rewrite-rs-controller.md (hunk 1 above)
Comment (yue9944882) [reply docs/java-controller-tutorial-rewrite-rs-controller.md]: fixed

File: docs/java-controller-tutorial-rewrite-rs-controller.md (hunk 2 above)
Comment (yue9944882) [reply docs/java-controller-tutorial-rewrite-rs-controller.md]: updated

**Pull request merged**